client/state/timezones/reducer.js

```diff
@@ -0,0 +1,48 @@
+/**
+ * External dependencies
+ */
+import { combineReducers } from 'redux';
+
+/**
+ * Internal dependencies
+ */
+import {
+	TIMEZONES_RECEIVE,
+	TIMEZONES_REQUEST,
+	TIMEZONES_REQUEST_FAILURE,
+	TIMEZONES_REQUEST_SUCCESS
+} from 'state/action-types';
+import { timezonesSchema } from './schema';
+import { createReducer } from 'state/utils';
+
+/**
+ * Collect and store timezones data
+ *
+ * @param  {Object} state - Current state
+ * @param  {Object} action - action payload
+ * @return {Object} updated state
+ */
+export const items = createReducer( {}, {
+	[ TIMEZONES_RECEIVE ]: ( state, {
+		rawOffsets,
+		timezonesByContinent,
+	} ) => ( { rawOffsets, timezonesByContinent } ),
+}, timezonesSchema );
+
+/**
+ * Track the timezones requesting process
```

Comment:
> Unless I'm misreading the wiring forrequestEligibility @dmsnell it doesn't look like we're checking for in-flight requests in the data-layer. Did you or @aduth have any plans for utilities there?

It's not there yet and it's waiting for the HTTP layer to merge in because that's honestly the place to perform network retries. For now though in the meantime if we are going to use the data layer I think we should be putting that logic into the API endpoint files themselves instead of in the components and reducers.